horizon_bsn/api/rest/neutron.py

```diff
@@ -0,0 +1,261 @@
+#
+#    (c) Copyright 2015 Hewlett-Packard Development Company, L.P.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""API over the neutron service.
+"""
+
+from django.utils.safestring import mark_safe
+from django.views import generic
+
+from openstack_dashboard.api.rest import urls
+from openstack_dashboard.api.rest import utils as rest_utils
+
+from horizon_bsn.api import neutron as bsnneutron
+
+from openstack_dashboard.api import heat
+from openstack_dashboard.api import neutron
+
+from horizon_bsn.content.connections.tabs import get_stack_topology
+
+import logging
+
+LOG = logging.getLogger(__name__)
+
+##################################################################
+# REACHABILITY TESTS
+##################################################################
+
+
+@urls.register
+class ReachabilityTest(generic.View):
+    """API for BSN Neutron Reachability Tests"""
+    url_regex = \
+        r'neutron/reachabilitytests/(?P<test_id>[^/]+|default)/$'
+
+    @rest_utils.ajax()
+    def patch(self, request, test_id):
+        result = bsnneutron.\
+            reachabilitytest_update(request, test_id, run_test=True)
+        return result
```

Comment:
I added a comment about returning objects vs json. The list functions (except router_rule since that returns a router) return json because upstream likes that better. The others are all objects.